Lib/random.py

```diff
@@ -375,14 +375,15 @@ def choices(self, population, weights=None, *, cum_weights=None, k=1):
         if cum_weights is None:
             if weights is None:
                 _int = int
+                n += 0.0    # convert to float for a small speed improvement
                 return [population[_int(random() * n)] for i in range(k)]
             cum_weights = list(_itertools.accumulate(weights))
         elif weights is not None:
             raise TypeError('Cannot specify both weights and cumulative weights')
         if len(cum_weights) != n:
             raise ValueError('The number of weights does not match the population')
         bisect = _bisect.bisect
-        total = cum_weights[-1]
+        total = cum_weights[-1] + 0.0
```

Comment:
Maybe repeat the comment? The code above can be changed independently, and the comment can be lost.

Reply:
Okay.